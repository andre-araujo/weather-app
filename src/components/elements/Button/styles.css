.button {
  border: 1px solid #333;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  padding: 0 30px;
  background: white;
  height: 60px;
  font-size: 20px;
  cursor: pointer;
  text-transform: uppercase;
  font-weight: bold;
  color: #333;
}

.button:hover {
  background: #333;
  color: #fff;
}
