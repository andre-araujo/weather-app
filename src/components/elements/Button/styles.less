@import '../../../styles/settings/colors.less';

.button {
  border: 1px solid @black10;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  padding: 0 30px;
  background: @white;
  height: 60px;
  font-size: 20px;
  cursor: pointer;
  text-transform: uppercase;
  font-weight: bold;
  color: @black10;
}

.button:hover {
  background: @black10;
  color: @white;
}
