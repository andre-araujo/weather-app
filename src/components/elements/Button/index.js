import React from 'react';
import { func, node, string } from 'prop-types';

import styles from './styles.less';

function Button(props) {
  const {
    type,
    onClick,
    children,
  } = props;

  return (
    <button data-test={props['data-test']} className={styles.button} type={type} onClick={onClick}>{children}</button>
  );
}

Button.defaultProps = {
  onClick: null,
  type: 'button',
  'data-test': null,
};

Button.propTypes = {
  onClick: func,
  type: string,
  'data-test': string,
  children: node.isRequired,
};

export default Button;
