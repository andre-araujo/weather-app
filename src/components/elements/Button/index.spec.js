import React from 'react';
import { shallow } from 'enzyme';

import Button from './index';

let props;

beforeEach(() => {
  props = {
    onClick: jest.fn(),
  };
});

describe('Button component', () => {
  it('should match snapshot', () => {
    expect(shallow(<Button {...props}>text</Button>)).toMatchSnapshot();
  });

  it('should call onClick prop when Button is clicked', () => {
    const button = shallow(<Button {...props}>text</Button>);
    button.simulate('click');

    expect(props.onClick).toHaveBeenCalled();
  });

  it('shouldn\'t call onClick prop when Button is not clicked', () => {
    shallow(<Button {...props}>text</Button>);

    expect(props.onClick).not.toHaveBeenCalled();
  });
});
