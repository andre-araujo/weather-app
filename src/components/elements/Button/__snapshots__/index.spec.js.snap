// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<button
  data-test={null}
  onClick={[Function]}
  type="button"
>
  text
</button>
`;
